### PR TITLE
Erlang 17.5 Compatibility

### PR DESCRIPTION
**Pull request by joedevivo**
This PR allows all projects to build in erlang 17.5 except for `chef-mover`.

`chef-mover` needs some attention to get that done, which it should be receiving shortly.

Since that's the case, these changes were all made to be compatible with R16B03-1 as well, and omnibus continues to build it that way.

### CODE REVIEW

File: src/bookshelf/rebar.config [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 {erl_opts, [debug_info,
             {parse_transform, lager_transform},
-            warnings_as_errors}].
+            warnings_as_errors]}.
```
Comment (marcparadise) [inline src/bookshelf/rebar.config]: How was that working before?

Comment (stevendanna) [inline src/bookshelf/rebar.config]: My guess was that it wasn't but we didn't notice because rebar.config.lock gets used instead for nearly all builds.

Comment (joedevivo) [inline src/bookshelf/rebar.config]: @stevendanna is right. rebar.config.lock was giving us a pass on this. Although I have no idea how the last person to generate it was successful

**Comment by stevendanna** [comment]
:+1:

**Pull request merged**